shopinvader_import_image/models/import_image.py

```diff
@@ -157,27 +174,19 @@ def _read_csv(self):
 
     def _get_lines(self):
         lines = []
-        with closing(io.BytesIO(self._read_csv())) as file_csv:
-            reader = csv_reader(file_csv, delimiter=self.csv_delimiter)
-            headers = next(reader, None)
-
-            if headers != self.csv_header.split(self.csv_delimiter):
-                raise exceptions.UserError(
-                    _("Invalid CSV file headers found! Expected: %s") % self.csv_header
-                )
+        mapping = {
+            "default_code": self.csv_column_default_code,
+            "tag_name": self.csv_column_tag_name,
+            "file_path": self.csv_column_file_path,
+        }
+        with closing(io.BytesIO(self._read_csv())) as binary_file:
+            csv_file = (line.decode("utf8") for line in binary_file)
+            reader = csv.DictReader(csv_file, delimiter=self.csv_delimiter)
             csv.field_size_limit(sys.maxsize)
-
             for row in reader:
-                if not row:
-                    continue
-                default_code, tag_name, file_path = row
-                lines.append(
-                    {
-                        "default_code": default_code,
-                        "tag_name": tag_name,
-                        "file_path": file_path,
-                    }
-                )
+                line = {key: row.get(column) for key, column in mapping.items()}
```

Comment:
do not use get and let it raise a KeyError, catch that exception and raise a UserError "CSV schema imcompatible" or something similar